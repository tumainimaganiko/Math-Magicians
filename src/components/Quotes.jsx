import { useEffect, useState } from "react";

export default function Quotes() {
  const [isLoading, setIsLoading] = useState(false);
  const [err, setErr] = useState(false);
  const [quotes, setQuotes] = useState([
    {
      quote: "",
      author: "",
      category: "",
    },
  ]);

  useEffect(() => {
    let ready = true;
    const fetchQuotes = async () => {
      setIsLoading(true);
      try {
        const response = await fetch("https://api.api-ninjas.com/v1/quotes", {
          method: "GET",
          headers: {
            "Content-Type": "application/json",
            "X-API-Key": "T4e57bd+nZWxSEgR4/XZNg==BOSUQttNXAPifpcW",
          },
        });
        const result = await response.json();
        if (ready) {
          setQuotes(...[], result);
        }
      } catch (error) {
        setErr(true);
      }
      setIsLoading(false);
    };
    fetchQuotes();

    return () => {
      ready = false;
    };
  }, [setIsLoading, setErr]);

  const wrapper = {
    display: "flex",
    "justify-content": "center",
    "align-items": "center",
    padding: "200px 30px",
  };

  return (
    <div className="wrapper" style={wrapper}>
      <div className="quote">
        {isLoading ? "Please Wait we are fetching Quotes" : ""}

        {!err
          ? `${quotes[0].quote}  ${quotes[0].author}`
          : "There is any error in fetching Data"}
      </div>
    </div>
  );
}
