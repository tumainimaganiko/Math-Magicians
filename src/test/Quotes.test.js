import renderer from 'react-test-renderer';
import Quotes from 'components/Quotes';

it('Render quotes', () => {
  const tree = renderer.create(<Quotes />).toJSON();
  expect(tree).toMatchSnapshot();
});
