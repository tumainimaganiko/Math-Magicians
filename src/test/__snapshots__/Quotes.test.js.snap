// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render quotes 1`] = `
<div
  className="wrapper"
  style={
    Object {
      "align-items": "center",
      "display": "flex",
      "justify-content": "center",
      "padding": "200px 30px",
    }
  }
>
  <div
    className="quote"
  >
      
  </div>
</div>
`;
